### PR TITLE
[Bug] Demo App AuthenticationHelper never called back in case of error

### PR DESCRIPTION
**Pull request by JoshuaLai**
## Purpose
Bug in our demo app, if the token fetcher fails for any reason we were never responding to the error, so the composite would freeze
Token struct decodable looks like it was also out of date. 
image

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/AuthenticationHelper.swift [hunk 1]
```diff
@@ -9,21 +9,29 @@ import AzureCommunicationCalling
 class AuthenticationHelper {
     static func getCommunicationToken(tokenUrl: URL) -> TokenRefresher {
         struct TokenResponse: Decodable {
+            let userAndToken: UserAndTokenResponse
+            let endpointUrl: String?
+        }
+        struct UserAndTokenResponse: Decodable {
             let token: String
+            let expiresOn: String?
+            let user: String?
```
Comment (kevinyulu) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/AuthenticationHelper.swift:18]: Should this user be an object type? I looked into the response json from the url. looks like this user is an object type. this is causing the decoding to fail from my side.

**Comment by AmyL219** [comment]
My token generation Url response is different from yours.So I couldn't login with token fetch url with the change. I think the response is generally apply to different Azure Function setting. For example, I only request token for `'voip'`

**Review by kevinyulu — DISMISSED** [review]
Checked with @JoshuaLai about the case of url response getting `401` due to authentication and page redirect before seeing the json response. I think this is outside the scope of this ticket and can be discussed separately. PR looks good.

**Comment by JoshuaLai** [comment]
> 



> My token generation Url response is different from yours.So I couldn't login with token fetch url with the change. I think the response is generally apply to different Azure Function setting. For example, I only request token for `'voip'` Screen Shot 2022-03-30 at 9 33 37 AM

Oh interesting I guess it makes sense that different urls return different bodies. Maybe I undo the change to the decodable object?

**Review by kevinyulu — APPROVED** [review]

**Review by palatter — APPROVED** [review]
Looks good and tested.

**Pull request merged**